src/helpers/get-directory-paths.js

```diff
@@ -1,6 +1,38 @@
-module.exports = filePaths => [...new Set(filePaths.map(path => {
-  const lastSlashIndex = path.lastIndexOf('/')
-  // if the filePath contains a '/', then return just the directory
-  // otherwise, return '' (i.e. this is the root)
-  return lastSlashIndex !== -1 ? path.slice(0, lastSlashIndex) : ''
-}))]
+// module.exports = filePaths => [...new Set(filePaths.map(path => {
+//   const lastSlashIndex = path.lastIndexOf('/')
+
+//   const split = path.split('/')
+//   console.log(split)
+//   // if the filePath contains a '/', then return just the directory
+//   // otherwise, return '' (i.e. this is the root)
+//   return lastSlashIndex !== -1 ? path.slice(0, lastSlashIndex) : ''
+// }))]
+
+const getDirectoryPaths = (filePaths) => {
+  const dirPaths = filePaths.map(path => {
+    // file = SomeOtherDir/AnotherDir/AnotherNestedDir/file.html
+    if (!path.includes('/')) return '' // if file is not in the directory, return '', as this is the root
+
+    const splitPaths = path.split('/') // => [ 'SomeOtherDir', 'AnotherDir', 'AnotherNestedDir', 'File1.html' ]
+    splitPaths.pop() // remove file.html
+
+    const allDirs = splitPaths.map((splitPath, index) => {
+      const newPath = joinPathParts(splitPaths, index)
+      return newPath
+    })
+
+    return allDirs
+  })
+
+  return [...new Set(dirPaths.flat())]
+}
+
+const joinPathParts = (pathsArray, end) => {
```

Comment:
This has some interesting behaviour, I'm not such a huge fan of leaking side-effects out of functions (the `index` in this case is changing in the function call, and then being used to dictate the logic outside of it)... we could rewrite this to something like:

```
// Returns all paths for a given array of paths, relative to the first path
// Input is assumed to be ordered, first element is the first path, second is a direct child, etc
// ['a', 'b', 'c'] -> ['a', 'a/b', 'a/b/c']
function allRelativePaths (paths) {
  const loop = paths.length
  let allPaths = []
  for (let i = 0; i < loop; i += 1) {
    const path = splitPaths.join('/')
    paths.push(path)
    splitPaths = splitPaths.slice(0, splitPaths.length - 1)
  }

  return allPaths.sort()
}
```

Which is not beautiful code, but should provide the same functionality, with no external side-effects. 

Instead of L19, we could then do `const allDIrs = allRelativePaths(splitPaths)`